langwatch/src/components/datasets/DatasetMapping.tsx

```diff
@@ -106,7 +110,7 @@ export const TRACE_MAPPINGS = {
     subkeys: (traces: TraceWithSpansAndAnnotations[], key: string) => {
       const spans = traces
         .flatMap((trace) => trace.spans ?? [])
-        .filter((span) => span.name === key);
+        .filter((span) => getSpanNameOrModel(span) === key);
```

Comment:
can you add the same helper on the trace details as well? this way we know they keep always aligned

feel free to do that in another PR if you want though, approving this